src/components/Account/Login.vue

```diff
@@ -0,0 +1,278 @@
+          </b-button>
+          <div class="split-line w-100 d-flex align-items-center">或</div>
+          <b-form class="w-100" @submit="handleLogin">
+            <b-form-group
+              v-for="(info, index) in loginInfos"
+              :key="index"
+              :label="info.name"
+              :label-for="info.name"
+              label-class="form-label"
+              class="form w-100"
+            >
+              <b-form-input
+                :id="info.name"
+                v-model="info.data"
+                :type="info.type"
+                :placeholder="info.name"
+                required
+                trim
+              />
+            </b-form-group>
+            <div class="w-100">
+              <b-form-checkbox size="sm" class="keep-login-button">
+                保持登入狀態
+              </b-form-checkbox>
+            </div>
+            <b-button type="submit" block style="background: #2353ff">
+              登入
+            </b-button>
+          </b-form>
+          <div class="option w-100 d-flex justify-content-between">
+            <b-link style="font-size: 14px; color: black"> 忘記密碼？ </b-link>
+            <b-link
+              style="font-size: 14px; color: #2353ff"
+              :to="{ name: 'SignUp', query: getRedirectPath }"
+            >
+              註冊新帳號
+            </b-link>
+          </div>
+        </div>
+      </b-col>
+    </b-row>
+  </b-container>
+</template>
+
+<script>
+import firebase from '@/utils/firebase'
+export default {
+  name: 'Login',
+  data() {
+    return {
+      slogans: [
+        '一個匯集資訊的同步協作平台',
+        '新聞的多元觀點與客觀事實',
+        '共同編輯新聞系統'
+      ],
+      thirdPartyLogins: [
+        // {
+        //   name: 'Facebook',
+        //   icon: 'facebook-circle'
+        // },
+        {
+          name: 'Google',
+          icon: 'google'
+        }
+      ],
+      loginInfos: [
+        {
+          type: 'text',
+          name: '帳號/Email',
+          data: ''
+        },
+        {
+          type: 'password',
+          name: '密碼',
+          data: ''
+        }
+      ]
+    }
+  },
+  computed: {
+    accountOrEmail: function() {
+      return this.loginInfos[0].data
+    },
+    password: function() {
+      return this.loginInfos[1].data
+    },
+    getRedirectPath: function() {
+      return this.$route.query.redirect ? { redirect: this.$route.query.redirect } : null
+    }
+  },
+  methods: {
+    async handleLogin() {
+      try {
+        await firebase.handleLogin(this.accountOrEmail, this.password)
+        this.$router.back()
+      } catch (error) {
+        this.loginInfos.forEach((info) => {
+          info.data = ''
+        })
+        console.error(error)
+        this.$bvModal.msgBoxOk(error.message)
+      }
+    },
+    thirdPartyLoginFunc(target) {
+      switch (target) {
+        // case 'Facebook':
+        //   return firebase.loginWithFacebook
+        case 'Google':
+          return firebase.loginWithGoogle
+      }
```

Comment:
Function binding missing.
When the function is called, [`this`](https://github.com/NCTU-SYNC/sync-frontend/blob/upgrade-tiptap/src/utils/firebase.js#L105) points to the Vue component instead of the firebase instance. Apparently, `this.auth` in the `Login` component is `undefined`.

Reply:
Check out [Javascript Function Binding](https://developer.mozilla.org/zh-TW/docs/Web/JavaScript/Reference/Global_Objects/Function/bind)